shared/src/http.rs

```diff
@@ -131,6 +135,104 @@ pub fn filter_hop_by_hop(headers: &mut HeaderMap, version: Version) -> &mut Head
     headers
 }
 
+/// Send a request to a single upstream with configurable timeout
+pub async fn send_to_upstream<C, B, E>(
+    client: &Client<C, B>,
+    upstream_url: &url::Url,
+    request: Request<B>,
+    timeout_secs: u64,
+) -> Result<Response<Bytes>, E>
+where
+    C: hyper_util::client::legacy::connect::Connect + Clone + Send + Sync + 'static,
+    B: hyper::body::Body + Send + Unpin + 'static,
+    B::Data: Send,
+    B::Error: std::error::Error + Send + Sync + 'static,
+    E: From<UpstreamError>,
+{
+    // Use host as identifier for error messages
+    let upstream_identifier = upstream_url.host_str().unwrap_or(upstream_url.as_str());
+
+    // Build the full upstream URI by combining base URL with request path
+    let path_and_query = request
+        .uri()
+        .path_and_query()
+        .map(|pq| pq.as_str())
+        .unwrap_or("/");
```

Comment:
this is different to the proxy. the proxy returns an error response if the path isn't valid instead of defaulting to `/` and just forwarding the request anyway.. is this intentional?

Reply:
Made the same behavior as proxy now